core/deployment/src/main/java/io/quarkus/deployment/pkg/steps/JvmStartupOptimizerArchiveBuildStep.java

```diff
@@ -44,22 +44,22 @@ public class JvmStartupOptimizerArchiveBuildStep {
 
     @BuildStep(onlyIf = AppCDSRequired.class)
     public void requested(PackageConfig packageConfig, OutputTargetBuildItem outputTarget,
-            BuildProducer<AppCDSRequestedBuildItem> producer)
+            BuildProducer<JvmStartupOptimizerArchiveRequestedBuildItem> producer)
             throws IOException {
         Path appCDSDir = outputTarget.getOutputDirectory().resolve("appcds");
         IoUtils.createOrEmptyDir(appCDSDir);
 
-        producer.produce(new AppCDSRequestedBuildItem(outputTarget.getOutputDirectory().resolve("appcds"),
+        producer.produce(new JvmStartupOptimizerArchiveRequestedBuildItem(outputTarget.getOutputDirectory().resolve("appcds"),
                 packageConfig.jar().appcds().useAot() ? JvmStartupOptimizerArchiveType.AOT
                         : JvmStartupOptimizerArchiveType.AppCDS));
     }
```

Comment:
This seems generic now but there are still references to AppCDS everywhere, including in the `@BuildStep` condition.

If it's AppCDS only, it feels weird to test for the type?

Reply:
Given we are making quite a lot of changes here, maybe we should clean up things once and for all.
We could deprecate what's existing but still take it into account to not break existing users and have properly named properties?

Now I will let you the judge of that. I'm not sure things are defined well enough for us to make the right decision.

Reply:
I'll think about it and come back after the weekend

Reply:
I actually prefer to do the config change in another step